helix-term/src/commands.rs

```diff
@@ -1767,8 +1768,15 @@ mod cmd {
         args: &[&str],
         _event: PromptEvent,
     ) -> anyhow::Result<()> {
-        let path = args.get(0).context("wrong argument count")?;
-        let _ = cx.editor.open(path.into(), Action::Replace)?;
+        use helix_core::path::expand_tilde;
+        let arg = args.get(0).context("wrong argument count")?;
+        let (path, pos) = args::parse_file(arg);
+        let _ = cx.editor.open(expand_tilde(&path), Action::Replace)?;
```

Comment:
There's no need to call `expand_tilde` anymore, the underlying bug got fixed

```suggestion
        let arg = args.get(0).context("wrong argument count")?;
        let (path, pos) = args::parse_file(arg);
        let _ = cx.editor.open(path.into(), Action::Replace)?;
```